File: VB/DXSample/ViewModels/MainViewModel.vb
Imports DevExpress.Mvvm
Imports System.Collections.ObjectModel

Namespace DXSample.ViewModels
	Public Class MainViewModel
		Inherits ViewModelBase

		Private ReadOnly Property MessageBoxService() As IMessageBoxService
			Get
				Return GetService(Of IMessageBoxService)()
			End Get
		End Property

		Public Property Items() As ObservableCollection(Of ButtonViewModel)
			Get
				Return GetProperty(Function() Items)
			End Get
			Set(ByVal value As ObservableCollection(Of ButtonViewModel))
				SetProperty(Function() Items, value)
			End Set
		End Property
		Public Sub New()
            Items = New ObservableCollection(Of ButtonViewModel)() From {
                New ButtonViewModel() With {
                    .Content = "A",
                    .Command = New DelegateCommand(Function() MessageBoxService.ShowMessage("A")),
                    .IsLeft = True
                },
                New ButtonViewModel() With {
                    .Content = "B",
                    .Command = New DelegateCommand(Function() MessageBoxService.ShowMessage("B")),
                    .IsLeft = True},
                New ButtonViewModel() With {
                    .Content = "C",
                    .Command = New DelegateCommand(Function() MessageBoxService.ShowMessage("C")),
                    .IsLeft = True},
                New ButtonViewModel() With {
                    .Content = "X",
                    .Command = New DelegateCommand(Function() MessageBoxService.ShowMessage("X"))
                },
                New ButtonViewModel() With {
                    .Content = "Y",
                    .Command = New DelegateCommand(Function() MessageBoxService.ShowMessage("Y"))},
                New ButtonViewModel() With {
                    .Content = "Z",
                    .Command = New DelegateCommand(Function() MessageBoxService.ShowMessage("Z"))},
                New ButtonViewModel() With {
                    .Content = "Clear",
                    .Command = New DelegateCommand(
                        Sub()
                            Items.Clear()
                            Items = New ObservableCollection(Of ButtonViewModel)(Items)
                        End Sub)
                },
                New ButtonViewModel() With {
                    .Content = "Add",
                    .Command = New DelegateCommand(
                        Sub()
                            Items.Add(New ButtonViewModel() With {
                                .Content = "New",
                                .IsLeft = Items.Count Mod 2 = 0
                            })
                            Items = New ObservableCollection(Of ButtonViewModel)(Items)
                        End Sub)
                }
            }
        End Sub
	End Class
End Namespace

File: CS/DXSample/ViewModels/MainViewModel.cs
using DevExpress.Mvvm;
using System.Collections.ObjectModel;

namespace DXSample.ViewModels {
    public class MainViewModel : ViewModelBase {

        private IMessageBoxService MessageBoxService { get { return GetService<IMessageBoxService>(); } }

        public ObservableCollection<ButtonViewModel> Items {
            get { return GetProperty(() => Items); }
            set { SetProperty(() => Items, value); }
        }
        public MainViewModel() {
            Items = new ObservableCollection<ButtonViewModel>() {
                new ButtonViewModel() {
                    Content = "A",
                    Command = new DelegateCommand(()=> {
                        MessageBoxService.ShowMessage("A");
                    }),
                    IsLeft = true,
                },
                new ButtonViewModel() {
                    Content = "B",
                    Command = new DelegateCommand(()=> {
                        MessageBoxService.ShowMessage("B");
                    }),
                    IsLeft = true,
                },
                new ButtonViewModel() {
                    Content = "C",
                    Command = new DelegateCommand(()=> {
                        MessageBoxService.ShowMessage("C");
                    }),
                    IsLeft = true,
                },
                new ButtonViewModel() {
                    Content = "X",
                    Command = new DelegateCommand(()=> {
                        MessageBoxService.ShowMessage("X");
                    }),
                },
                new ButtonViewModel() {
                    Content = "Y",
                    Command = new DelegateCommand(()=> {
                        MessageBoxService.ShowMessage("Y");
                    }),
                },
                new ButtonViewModel() {
                    Content = "Z",
                    Command = new DelegateCommand(()=> {
                        MessageBoxService.ShowMessage("Z");
                    }),
                },
                new ButtonViewModel() {
                    Content = "Clear",
                    Command = new DelegateCommand(()=> {
                        this.Items.Clear();
                        Items = new ObservableCollection<ButtonViewModel>(Items);
                    }),
                },
                new ButtonViewModel() {
                    Content = "Add",
                    Command = new DelegateCommand(()=> {
                        Items.Add(new ButtonViewModel() {
                            Content = "New",
                            IsLeft = this.Items.Count % 2 == 0
                        });
                        Items = new ObservableCollection<ButtonViewModel>(Items);
                    }),
                },
            };
        }
    }
}